server/data/prisonApiClient.ts

```diff
@@ -29,15 +37,140 @@ export default class PrisonApiClient extends AbstractHmppsRestClient {
     return this.put({ path: '/api/users/me/activeCaseLoad', data, authToken: user.token })
   }
 
-  async searchActivityLocations(
-    prisonCode: string,
+  // eslint-disable-next-line  @typescript-eslint/no-explicit-any
+  async searchActivityLocations(prisonCode: string, date: string, period: string, user: ServiceUser): Promise<any[]> {
```

Comment:
Unless there was a reason for changing this??

Reply:
Yeah this didnt work for me - the actual response from dev doesnt include certain required attributes - e.g. agencyId: string; currentOccupancy?: number; description: string and more. This is only highlighted because i've used real responses from dev to construct my tests. Its compile time checking so if i removed the test - or used unrealistic api responses that included the required attributes - we would be good, but that feels worse than using any.

Reply:
As discussed yesterday i have created versions of the api types with very minor changes that are slightly more lenient - making a couple of attributes optional and changing enum types to strings. This has allowed me to replace the 'any' types.